org.eclipse.winery.crawler/src/main/java/org/eclipse/winery/crawler/chefcookbooks/ChefCookbookAnalyzer.java

```diff
@@ -0,0 +1,497 @@
+     * @return Returns parse result with added information about installed and required packages.
+     */
+    private CookbookParseResult returnComponentTypesFromRecipes(CookbookParseResult cookbookConfigs) {
+        if (cookbookConfigs.isInRecursiveTransformation() == false ) {
+            String defaultRecipePath = cookbookConfigs.getCookbookPath() + "/recipes" + "/default.rb";
+            System.out.println("Parsing Recipes: default.rb ...");
+            return getParseResultFromFile(cookbookConfigs, defaultRecipePath);
+        } else {
+            List<CookbookParseResult> parseResultList = cookbookConfigs.getListOfConfigsInOwnParseresult();
+
+            CookbookParseResult filteredParseResult;
+
+            List<ChefCookbookConfiguration> processedCookbookConfigs = new LinkedList<>();
+
+            // If parse result have multiple cookbook configurations, the AST is visited for each configuration.
+            for (int countConfigs = 0; countConfigs < parseResultList.size(); countConfigs++) {
+                //Parse result with a single configuration
+                filteredParseResult = parseResultList.get(countConfigs);
+                System.out.println(filteredParseResult.getAllConfigsAsList().get(0).getRunlist());
+                List<String> runlist = filteredParseResult.getAllConfigsAsList().get(0).getRunlist();
+                for ( int i = 0; i < runlist.size() ; i++) {
+                    String RecipePath = cookbookConfigs.getCookbookPath() + "/recipes" + "/" + runlist.get(i);
+                    System.out.println("Parsing Recipes: " + runlist.get(i) + "...");
+                    filteredParseResult = getParseResultFromFile(filteredParseResult, RecipePath);
+                }
+                
+                processedCookbookConfigs.add(filteredParseResult.getAllConfigsAsList().get(0));
+                // Reset
+                filteredParseResult.clearConfigurations();
+            }
+            cookbookConfigs.replaceCookbookConfigs(processedCookbookConfigs);
+            
+            return cookbookConfigs;
+        }
+        
+    }
+
+    /**
+     * This method parses ruby files from chef cookbooks for deployment architecture information.
+     * @param cookbookConfigs Present parsing result where new information are added.
+     * @param rbFilePath Path to file that should be parsed.
+     * @return New deployment architecture information from the parsed file.
+     * @throws IOException if the file can not be read.
+     */
+    public static CookbookParseResult getParseResultFromFile(CookbookParseResult cookbookConfigs, String rbFilePath) {
+        CookbookParseResult extractedCookbookConfigs;
+        if (CrawledCookbooks.fileExists(rbFilePath)) {
+            //RubyCodeHelper.prepareCodeFromFile(rbFilePath);
+            try {
+                File file = RubyCodeHelper.prepareCodeFromFile(rbFilePath);
```

Comment:
File is only a representation of a pathname and not a Stream. It does not implement a close() method or the interface AutoCloseable.

Reply:
considering that `prepareCodeFromFile` already transforms the code you want to look at into a `String` you could sidestep disk IO by using `CharStreams.fromString` instead of `fromFileName`.

In addition to that, you might want to look into using `Path` and `CharStreams.fromPath` as the newer and cleaner way to deal with files if you can't help but putting the code to compile into a file on disk.
Note that the comment by Lukas very much applies to the internals of `RubyCodeHelper::prepareCodeFromFile`